javatests/arcs/android/storage/database/AndroidSqliteDatabaseFactoryTest.kt

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.android.storage.database
+
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import arcs.core.storage.database.DatabaseFactory
+import com.google.common.truth.Truth.assertThat
+import kotlinx.coroutines.async
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.util.Random
+
+@RunWith(AndroidJUnit4::class)
+class AndroidSqliteDatabaseFactoryTest {
+    private lateinit var factory: DatabaseFactory
+    private lateinit var random: Random
+
+    @Before
+    fun setUp() {
+        factory = AndroidSqliteDatabaseFactory(ApplicationProvider.getApplicationContext())
+        random = Random(System.currentTimeMillis())
+    }
+
+    @Test
+    fun getDatabase() = runBlockingTest {
+        val database = factory.getDatabase("foo", true)
+        assertThat(database).isInstanceOf(DatabaseImpl::class.java)
+    }
+
+    @Test
+    fun getDatabases() = runBlockingTest {
+        val databaseFoo = factory.getDatabase("foo", true)
+        assertThat(databaseFoo).isInstanceOf(DatabaseImpl::class.java)
+
+        val databaseBar = factory.getDatabase("bar", true)
+        assertThat(databaseBar).isInstanceOf(DatabaseImpl::class.java)
+        assertThat(databaseFoo !== databaseBar).isTrue()
```

Comment:
```
assertThat(databaseFoo).isNotSameInstanceAs(databaseBar)
```

Reply:
(And throughout file)

Reply:
Ah, didn't realize that was a thing, thanks!